tests/cases/conformance/salsa/typeFromJSInitializer2.ts

```diff
@@ -0,0 +1,18 @@
+// @allowJs: true
+// @checkJs: true
+// @noEmit: true
+// @strictNullChecks: true
```

Comment:
for ease of reading, it would be nice to have a strictNullChecks: false version of these tests too. it would also help in case we decide to change this behaviour later.